crypto/s2n_ecc_evp.c

```diff
@@ -185,6 +191,31 @@ static int s2n_ecc_evp_compute_shared_secret(EVP_PKEY *own_key, EVP_PKEY *peer_p
     return 0;
 }
 
+static bool s2n_ecc_evp_is_available(uint16_t query_iana_id) {
+    switch (query_iana_id) {
+    case TLS_EC_CURVE_SECP_521_R1:
+        return !s2n_is_in_fips_mode();
+    case TLS_EC_CURVE_SECP_256_R1:
+    case TLS_EC_CURVE_SECP_384_R1:
+        return true;
+    case TLS_EC_CURVE_ECDH_X25519:
+        return s2n_is_evp_apis_supported();
```

Comment:
Could some of these be removed with this PR now? https://github.com/awslabs/s2n/blob/71cfa3f1c316810404a0cc0ca4533dacb5b41491/tls/s2n_ecc_preferences.c#L29

Reply:
we still have a s2n_ecc_curve_x25519 structure even EVP_APIS_SUPPORTED is disable.

https://github.com/awslabs/s2n/blob/71cfa3f1c316810404a0cc0ca4533dacb5b41491/crypto/s2n_ecc_evp.c#L89